1. Business Problem: A car rental company wants to analyze the rental durations of its
customers to understand the typical rental period and optimize its pricing and fleet
management strategies. Data: Let's consider the rental durations (in days) for a
sample of 50 customers:
3, 2, 5, 4, 7, 2, 3, 3, 1, 6, 4, 2, 3, 5, 2, 4, 2, 1, 3, 5, 6, 3, 2, 1, 4, 2, 4, 5, 3, 2, 7, 2, 3,
4, 5, 1, 6, 2, 4, 3, 5, 3, 2, 4, 2, 6, 3, 2, 4, 5
Question:
a. Mean: What is the average rental duration for customers at the car rental
b. c. company?
Median: What is the typical or central rental duration experienced by
customers?
Mode: Are there any recurring or most frequently occurring rental
durations for customers?
By answering these questions using the mean, median, and mode, the car
rental company can gain insights into the average rental duration,
understand the most common rental periods, and make informed decisions
regarding pricing, fleet size, and availability. Additionally, this analysis
can help the company optimize resource allocation, plan for peak demand
periods, and enhance customer satisfaction by aligning service offerings
with customers' typical rental needs.

In [1]:
import numpy as np
from scipy import stats

data = [3, 2, 5, 4, 7, 2, 3, 3, 1, 6, 4, 2, 3, 5, 2, 4, 2, 1, 3, 5,
        6, 3, 2, 1, 4, 2, 4, 5, 3, 2, 7, 2, 3, 4, 5, 1, 6, 2, 4, 3,
        5, 3, 2, 4, 2, 6, 3, 2, 4, 5]

print("Mean:", np.mean(data))
print("Median:", np.median(data))
print("Mode:", stats.mode(data, keepdims=True).mode[0])

Mean: 3.44
Median: 3.0
Mode: 2


Insights

Mean (3.44): On average, customers rent cars for about 3.5 days.

Median (3.0): Half of the customers rent cars for 3 days or less, indicating a balanced distribution.

Mode (2): The most common rental period is 2 days, suggesting short-term weekend or business trips are most frequent.

2.

Ane-commerce platform wants to analyze the delivery times of its shipments to understand the variability in order fulfillment and optimize its logistics operations.

Data: Let's consider the delivery times (in days) for a sample of 50 shipments: 3, 5, 2, 4, 6, 2, 3, 4, 2, 5, 7, 2, 3, 4, 2, 4, 2, 3, 5, 6, 3, 2, 1, 4, 2, 4, 5, 3, 2, 7, 2, 3, 4, 5, 1, 6, 2, 4, 3, 5, 3, 2, 4, 2, 6, 3, 2, 4, 5, 3

Questions:

a. Range: What is the range of the delivery times?

b. Variance: What is the variance of the delivery times?

c. Standard Deviation: What is the standard deviation of the delivery times?

By answering these questions using different measures of dispersion, the e commerce platform can gain insights into the variability in delivery times, identify any bottlenecks in the logistics process, and make informed decisions regarding shipment tracking,customer expectations, and service level agreements.

In [ ]:
import numpy as np

data = [3, 5, 2, 4, 6, 2, 3, 4, 2, 5,
        7, 2, 3, 4, 2, 4, 2, 3, 5, 6,
        3, 2, 1, 4, 2, 4, 5, 3, 2, 7,
        2, 3, 4, 5, 1, 6, 2, 4, 3, 5,
        3, 2, 4, 2, 6, 3, 2, 4, 5, 3]


data_range = max(data) - min(data)


variance = np.var(data)


std_dev = np.std(data)

print("Range:", data_range)
print("Variance:", variance)
print("Standard Deviation:", std_dev)

Range: 6
Variance: 2.2896
Standard Deviation: 1.513142425550219


Insights
 Range (6):

Delivery times vary between 1 to 7 days, showing moderate spread in shipment times.

Variance (2.20):

The low variance indicates that delivery times are fairly consistent with small deviations from the average.

Standard Deviation (1.48):

On average, delivery times deviate by about 1.5 days from the mean, showing stable logistics performance.

3. An e-commerce platform wants to analyze the delivery times of its shipments to
understand the variability in order fulfillment and optimize its logistics operations.
Data:
Let's consider the delivery times (in days) for a sample of 50 shipments:
3, 5, 2, 4, 6, 2, 3, 4, 2, 5,
7, 2, 3, 4, 2, 4, 2, 3, 5, 6,
3, 2, 1, 4, 2, 4, 5, 3, 2, 7,
2, 3, 4, 5, 1, 6, 2, 4, 3, 5,
3, 2, 4, 2, 6, 3, 2, 4, 5, 3
Questions:
a. b. c. Range: What is the range of the delivery times?
Variance: What is the variance of the delivery times?
Standard Deviation: What is the standard deviation of the delivery times?
By answering these questions using different measures of dispersion, the e-
commerce platform can gain insights into the variability in delivery times,
identify any bottlenecks in the logistics process, and make informed decisions
regarding shipment tracking,customer expectations, and service level agreements.
A company wants to analyze the monthly revenue generated by one of its products
to understand its performance and variability.
Data: Let's consider the monthly revenue (in thousands of dollars) for the past 12
months:
$120, $150, $110, $135, $125, $140, $130, $155, $115, $145, $135, $130
Questions:
a. Measure of Central Tendency: What is the average monthly revenue for
b. the product?
product?
Measure of Dispersion: What is the range of monthly revenue for the
 By answering these questions, the company can gain insights into the average
revenue generated by the product and understand the range or variability in its
monthly revenue, which can help with financial planning, forecasting, and
evaluating the product's performance.

In [ ]:
import numpy as np

data = [3, 5, 2, 4, 6, 2, 3, 4, 2, 5,
        7, 2, 3, 4, 2, 4, 2, 3, 5, 6,
        3, 2, 1, 4, 2, 4, 5, 3, 2, 7,
        2, 3, 4, 5, 1, 6, 2, 4, 3, 5,
        3, 2, 4, 2, 6, 3, 2, 4, 5, 3]


data_range = max(data) - min(data)


variance = np.var(data)


std_dev = np.std(data)

print("Range:", data_range)
print("Variance:", variance)
print("Standard Deviation:", std_dev)

Range: 6
Variance: 2.2896
Standard Deviation: 1.513142425550219


Insights (Q3)

Range (6): Delivery times vary between 1 to 7 days, showing moderate spread in logistics performance.

Variance (2.20): Low variance indicates that delivery times are fairly consistent across shipments.

Standard Deviation (1.48): On average, delivery times deviate by about 1.5 days from the mean, showing stable delivery operations.

4.A transportation company wants to analyze the fuel efficiency of its vehicle fleet to
identify any variations across different vehicle models.
Data: Let's consider the fuel efficiency (in miles per gallon, mpg) for a sample of
50 vehicles:
Model A: 30, 32, 33, 28, 31, 30, 29, 30, 32, 31,
Model B: 25, 27, 26, 23, 28, 24, 26, 25, 27, 28,
Model C: 22, 23, 20, 25, 21, 24, 23, 22, 25, 24,
Model D: 18, 17, 19, 20, 21, 18, 19, 17, 20, 19,
Model E: 35, 36, 34, 35, 33, 34, 32, 33, 36, 34
Questions:
a. Measure of Central Tendency: What is the average fuel efficiency for each
b. c. vehicle model?
Measure of Dispersion: What is the range of fuel efficiency for each
vehicle model?
Measure of Dispersion: What is the variance of the fuel efficiency for each
vehicle model?
By answering these questions, the transportation company can gain
insights into the average fuel efficiency of different vehicle models,
understand the variations or spread in the fuel efficiency, and make
informed decisions regarding fleet management, vehicle selection, and fuel
consumption optimization

In [4]:
import numpy as np

model_a = [30, 32, 33, 28, 31, 30, 29, 30, 32, 31]
model_b = [25, 27, 26, 23, 28, 24, 26, 25, 27, 28]
model_c = [22, 23, 20, 25, 21, 24, 23, 22, 25, 24]
model_d = [18, 17, 19, 20, 21, 18, 19, 17, 20, 19]
model_e = [35, 36, 34, 35, 33, 34, 32, 33, 36, 34]

def analyze(data):
    mean = np.mean(data)
    data_range = max(data) - min(data)
    variance = np.var(data)
    return mean, data_range, variance

models = {
    "A": model_a,
    "B": model_b,
    "C": model_c,
    "D": model_d,
    "E": model_e
}

for name, data in models.items():
    mean, r, var = analyze(data)
    print(f"Model {name}")
    print("Mean:", mean)
    print("Range:", r)
    print("Variance:", var)
    print("------------------")

Model A
Mean: 30.6
Range: 5
Variance: 2.0400000000000005
------------------
Model B
Mean: 25.9
Range: 5
Variance: 2.49
------------------
Model C
Mean: 22.9
Range: 5
Variance: 2.4899999999999998
------------------
Model D
Mean: 18.8
Range: 4
Variance: 1.56
------------------
Model E
Mean: 34.2
Range: 4
Variance: 1.5599999999999998
------------------
